google_compute_engine/network_utils.py

```diff
@@ -38,11 +44,16 @@ def _CreateInterfaceMap(self):
       dict, string MAC addresses mapped to the string network interface name.
     """
     interfaces = {}
-    for interface in os.listdir('/sys/class/net'):
+    for interface in netifaces.interfaces():
       try:
-        mac_address = open(
-            '/sys/class/net/%s/address' % interface).read().strip()
-      except (IOError, OSError) as e:
+        mac_address = (
+            netifaces.ifaddresses(interface)[netifaces.AF_LINK][0]['addr'])
+        # In some systems this field can come with an empty string or with the
+        # name of the interface when there is no MAC address associated with it.
+        # Check the regex to be sure
+        if not MAC_REGEX.match(mac_address):
+            raise Exception('No MAC Address')
+      except Exception as e:
```

Comment:
Can we make this exception any more specific to the possible excepts netifaces might throw?